tern/formats/cyclonedx/cyclonedx_common.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2021 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: BSD-2-Clause
+
+"""
+Common functions that are useful for CycloneDX document creation
+"""
+
+import datetime
+import uuid
+from tern.utils import general
+
+
+###################
+# General Helpers #
+###################
+
+
+# document level tool information
+metadata_tool = {
+    'vendor': 'Tern Tools',
+    'name': 'Tern',
+    'version': general.get_git_rev_or_version()[1]
+}
+
+
+# keys are what Tern uses, values what CycloneDX uses
+hash_type_mapping = {
+    'md5': 'MD5',
+    'sha1': 'SHA-1',
+    'sha256': 'SHA-256',
+}
+
+
+pkg_format_purl_type_mapping = {
+    'deb': 'deb',
+    'rpm': 'rpm',
+    'apk': 'apk',
+    'pip': 'pip',
+    'gem': 'gem',
+    'npm': 'npm',
+    'go.mod': 'go',
+}
+
+
+purl_types_with_namespaces = [
+    'deb',
+    'rpm',
+    'apk',
+]
+
+
+# map Tern OS guesses to package URL namespace
+os_guess_purl_namespace_mapping = {
+    'debian': 'debian',
+    'ubuntu': 'ubuntu',
+    'alpine linux': 'alpine',
+    'centos': 'centos',
+    'fedora': 'fedora',
+    'opensuse': 'opensuse',
+    'rhel': 'rhel',
+    # Arch Linux
+    # Photon
+}
+
+
+def get_serial_number():
+    ''' Return a randomly generated CycloneDX BOM serial number '''
+    return 'urn:uuid:' + str(uuid.uuid4())
+
+
+def get_timestamp():
+    ''' Return a timestamp suitable for the BOM timestamp '''
+    return datetime.datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ')
+
+
+def get_hash(checksum_type, checksum):
+    ''' Return a CycloneDX hash object from Tern checksum values '''
+    hash_algorithm = hash_type_mapping.get(checksum_type, None)
+    return None if hash_algorithm is None else {'alg': hash_algorithm, 'content': checksum}
+
+
+def get_property(name, value):
+    ''' Return a CycloneDX property object '''
+    return {'name': name, 'value': value}
+
+
+def get_purl_type(pkg_format):
+    return pkg_format_purl_type_mapping.get(pkg_format.lower())
+
+
+def get_purl_namespace(os_guess, pkg_format):
+    if pkg_format in purl_types_with_namespaces:
+        for os in os_guess_purl_namespace_mapping:
+            if os_guess.lower().startswith(os):
+                return os_guess_purl_namespace_mapping.get(os)
+    return None
+
+
+def get_os_guess(image_obj):
+    for layer in image_obj.layers:
```

Comment:
You can just return `image_obj.layers[0].os_guess` if it's not empty.

Reply:
I thought that _might_ have been the case. Thanks